core/che-core-db/src/main/java/org/eclipse/che/core/db/JNDIDataSourceFactory.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.core.db;
+
+
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
+import org.apache.tomcat.dbcp.dbcp2.BasicDataSourceFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.naming.Context;
+import javax.naming.Name;
+import javax.naming.spi.ObjectFactory;
+import java.util.Hashtable;
+import java.util.Properties;
+
+public class JNDIDataSourceFactory implements ObjectFactory {
```

Comment:
>So you could directly use that JNDI factory as well.

Directly I can't. But I use it indirectly over ```BasicDataSourceFactory.createDataSource``` method. 

>Also do you have considered ```HikariCP```

Yes. 

Procs
- very tasty on high load. 

Cons
- Our load is not that high
- We have no experience with it.
- We need a CQ for it.
- I wrote a note about it to @gorkem. But looks like it's not that important at this stage